### PR TITLE
Move TCL test unit/slowlog to Go case

### LINKED ISSUE #861: Move TCL test unit/slowlog to Go case

### PR DESCRIPTION
**Pull request by tanruixiang**
It closes #861

### CODE REVIEW

**Review by tisonkun — COMMENTED** [review]
I think for most commands `go-redis` should provide encapsulation so that you don't have to use `rdb.Do` but `rdb.<CommandName>`.

**Comment by tanruixiang** [comment]
> I think for most commands `go-redis` should provide encapsulation so that you don't have to use `rdb.Do` but `rdb.<CommandName>`.

Ok. I will change it.

File: tests/gocase/unit/slowlog/slowlog_test.go [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+	"strings"
+	"testing"
+
+	"github.com/apache/incubator-kvrocks/tests/gocase/util"
+	"github.com/go-redis/redis/v9"
+	"github.com/stretchr/testify/require"
+)
+
+func TestSlowlog(t *testing.T) {
+	srv := util.StartServer(t, map[string]string{})
```
Comment (tisonkun) [inline tests/gocase/unit/slowlog/slowlog_test.go]: Default config contains: `slowlog-log-slower-than 1000000`.

File: tests/gocase/unit/slowlog/slowlog_test.go [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+	ctx := context.Background()
+	rdb := srv.NewClient()
+	defer func() { require.NoError(t, rdb.Close()) }()
+
+	t.Run("SLOWLOG - check that it starts with an empty log", func(t *testing.T) {
+		require.EqualValues(t, 0, rdb.Do(ctx, "slowlog", "len").Val())
+	})
+
+	t.Run("SLOWLOG - only logs commands taking more time than specified", func(t *testing.T) {
+		require.NoError(t, rdb.Do(ctx, "config", "set", "slowlog-log-slower-than", 100000).Err())
```
Comment (tisonkun) [inline tests/gocase/unit/slowlog/slowlog_test.go]: use `rdb.ConfigSet`

File: tests/gocase/unit/slowlog/slowlog_test.go [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+		require.NoError(t, rdb.Do(ctx, "config", "set", "slowlog-log-slower-than", 0).Err())
+		require.NoError(t, rdb.Do(ctx, "config", "set", "slowlog-max-len", 10).Err())
+		for i := 0; i < 100; i++ {
+			require.NoError(t, rdb.Ping(ctx).Err())
+		}
+		require.EqualValues(t, 10, rdb.Do(ctx, "slowlog", "len").Val())
+	})
+
+	t.Run("SLOWLOG - GET optional argument to limit output len works", func(t *testing.T) {
+		require.EqualValues(t, 5, reflect.ValueOf(rdb.Do(ctx, "slowlog", "get", 5).Val()).Len())
```
Comment (tisonkun) [inline tests/gocase/unit/slowlog/slowlog_test.go]: use `rdb.SlowLogGet` and assert the return `[]SlowLog` array.

File: tests/gocase/unit/slowlog/slowlog_test.go (hunk 2 above)
Comment (tisonkun) [reply tests/gocase/unit/slowlog/slowlog_test.go]: ditto others

File: tests/gocase/unit/slowlog/slowlog_test.go [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+	})
+
+	t.Run("SLOWLOG - RESET subcommand works", func(t *testing.T) {
+		require.NoError(t, rdb.Do(ctx, "config", "set", "slowlog-log-slower-than", 100000).Err())
+		require.NoError(t, rdb.Do(ctx, "slowlog", "reset").Err())
+		require.EqualValues(t, 0, rdb.Do(ctx, "slowlog", "len").Val())
+	})
+
+	t.Run("SLOWLOG - logged entry sanity check", func(t *testing.T) {
```
Comment (tisonkun) [inline tests/gocase/unit/slowlog/slowlog_test.go:70]: use `rdb.SlowLogGet` and assert the return `[]SlowLog` array.

File: tests/gocase/unit/slowlog/slowlog_test.go [hunk 5]
```diff
@@ -0,0 +1,170 @@
...
+
+	t.Run("SLOWLOG - Rewritten commands are logged as their original command", func(t *testing.T) {
+		require.NoError(t, rdb.Do(ctx, "config", "set", "slowlog-log-slower-than", 0).Err())
+		// Test rewriting client arguments
+		require.NoError(t, rdb.SAdd(ctx, "set", "a", "b", "c", "d", "e").Err())
+		require.NoError(t, rdb.Do(ctx, "slowlog", "reset").Err())
+
+		// SPOP is rewritten as DEL when all keys are removed
+		require.NoError(t, rdb.SPopN(ctx, "set", 10).Err())
+		cmd := rdb.Do(ctx, "slowlog", "get")
```
Comment (tisonkun) [inline tests/gocase/unit/slowlog/slowlog_test.go]: ditto `SlogLogGet`

**Review by tisonkun — CHANGES_REQUESTED** [review]
Generally look good. Comments inline to write tidy code.

**Comment by tisonkun** [comment]
@tanruixiang the next time you can ping me when the PR is ready to review. Otherwise I don't get a notification.

**Comment by tanruixiang** [comment]
> @tanruixiang the next time you can ping me when the PR is ready to review. Otherwise I don't get a notification.

@tisonkun  Thanks for your reminder. I'm guessing you're on vacation, and I'm afraid of disturbing you, so I didn't click the request for review button. ❤️

**Review by tisonkun — APPROVED** [review]

File: tests/gocase/unit/slowlog/slowlog_test.go [hunk 6]
```diff
@@ -39,13 +39,13 @@ func TestSlowlog(t *testing.T) {
...
 
 	t.Run("SLOWLOG - check that it starts with an empty log", func(t *testing.T) {
-		require.EqualValues(t, 0, rdb.Do(ctx, "slowlog", "len").Val())
+		require.EqualValues(t, 0, len(rdb.SlowLogGet(ctx, -1).Val()))
 	})
 
 	t.Run("SLOWLOG - only logs commands taking more time than specified", func(t *testing.T) {
 		require.NoError(t, rdb.ConfigSet(ctx, "slowlog-log-slower-than", "100000").Err())
 		require.NoError(t, rdb.Ping(ctx).Err())
-		require.EqualValues(t, 0, rdb.Do(ctx, "slowlog", "len").Val())
```
Comment (tisonkun) [inline tests/gocase/unit/slowlog/slowlog_test.go]: Although I think using `SLOWLOG LEN` fits the original code and covers most functionality, these tests can be OK to merge from my perspective.

Merging...

**Comment by tisonkun** [comment]
@tanruixiang Thank you!

**Pull request merged**